app/javascript/packs/pages/binx_admin.js

```diff
@@ -2,98 +2,120 @@ import log from "../utils/log";
 import moment from "moment-timezone";
 import LoadFancySelects from "../utils/LoadFancySelects";
 
-window.BinxAdmin = class BinxAdmin {
-  init() {
-    // If there is an element on the page with id pageContainerFluid, make the page container full width
-    if ($("#pageContainerFluid").length) {
-      $("#admin-content > .receptacle").css("max-width", "100%");
-    } else {
-      // Also, make full-screen-table receptacle max-width
-      $(".full-screen-table")
-        .parents(".receptacle")
-        .css("max-width", "100%");
-    }
-    if ($(".calendar-box")[0]) {
-      window.binxAdmin.setCustomGraphStartAndSlide();
-      this.changeGraphCalendarBox();
-    }
-    if ($("#use_image_for_display").length > 0) {
-      this.useBikeImageForDisplay();
-    }
-    if ($("#admin-locations-fields").length > 0){
-    this.adminLocations()
-    }
-    // Enable bootstrap custom file upload boxes
-    binxApp.enableFilenameForUploads();
-    LoadFancySelects();
-  }
-
-  changeGraphCalendarBox() {
-    $("select#graph_date_option").on("change", e => {
-      e.preventDefault();
-      this.setCustomGraphStartAndSlide();
-    });
-  }
+function BinxAdmin() {
+  return {
+    init() {
+      // If there is an element on the page with id pageContainerFluid, make the page container full width
+      if ($("#pageContainerFluid").length) {
+        $("#admin-content > .receptacle").css("max-width", "100%");
+      } else {
+        // Also, make full-screen-table receptacle max-width
+        $(".full-screen-table")
+          .parents(".receptacle")
+          .css("max-width", "100%");
+      }
+      if ($(".calendar-box")[0]) {
+        this.setCustomGraphStartAndSlide();
+        this.changeGraphCalendarBox();
+      }
+      if ($("#use_image_for_display").length > 0) {
+        this.useBikeImageForDisplay();
+      }
+      if ($("#admin-locations-fields").length > 0) {
+        this.adminLocations();
+      }
+      // Enable bootstrap custom file upload boxes
+      binxApp.enableFilenameForUploads();
+      LoadFancySelects();
+    },
 
-  startGraphTimeSet() {
-    let graphSelected = $("select#graph_date_option")[0].value.split(",");
-    let amount = Number(graphSelected[0]);
-    let unit = graphSelected[1];
-    $("#start_at").val(
-      moment()
-        .subtract(amount, unit)
-        .format("YYYY-MM-DDTHH:mm")
-    );
-  }
+    changeGraphCalendarBox() {
+      $("select#graph_date_option").on("change", e => {
+        e.preventDefault();
+        this.setCustomGraphStartAndSlide();
+      });
+    },
 
-  setCustomGraphStartAndSlide() {
-    if ($("select#graph_date_option")[0].value === "custom") {
-      $(".calendar-box").slideDown();
-    } else {
-      $(".calendar-box").slideUp();
-      this.startGraphTimeSet();
-    }
-  }
+    startGraphTimeSet() {
+      let graphSelected = $("select#graph_date_option")[0].value.split(",");
+      let amount = Number(graphSelected[0]);
+      let unit = graphSelected[1];
```

Comment:
Any objection to defaulting to `const`?

Reply:
Is there any difference? Can you mutate `const`?

Reply:
That's the main difference—you can't re-assign `const` (so the binding is immutable, but any non-primitive object bound to the variable can still be mutated). Scoping is the same. `const` provides marginally more assurance about state, but...the emphasis there is on _marginal_, so I don't feel too strongly about it.

Reply:
I’m happy to default to const, seems good.

Reply:
I switched these to `const`, but didn't switch the `for (let name of Array.from(names))` - which might be rewriteable in some other way, but I worry about reassigning the variable? Let me know if you think I should switch those too.

Reply:
> for (let name of Array.from(names))

I wasn't sure about this myself, but it looks like `const` work as expected in for-of / for-in loops but not in imperative for loops:

```js
> arr = [1,2,3,4]

> for (let num in arr) { console.log(num) }
0
1
2
3

> for (const num in arr) { console.log(num) }
0
1
2
3

> for (const i in arr) { console.log(arr[i]) }
1
2
3
4

> for (const i = 0; i < arr.length; i++) { console.log(arr[i]) }
1
TypeError: Assignment to constant variable.
```

Reply:
Yeah. apparently because ++ is reassignment. I'm happy with how it is now, since `const` as an iterator seems weird to me - but I'm happy to swap it out if you think we should use exclusively `const`